app/models/goal.py

```diff
@@ -1,5 +1,21 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
 
+
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    tasks: Mapped[list["Task"]] = relationship(back_populates="goal")
+
+    def to_dict(self):
+        return {
+            "goal": {
+                "id": self.id,
+                "title": self.title,
+                }
+        }
+
+    @classmethod
+    def from_dict(cls, goal_data):
+        new_goal = cls(title=goal_data["title"])
+        return new_goal
```

Comment:
I like how you just return a dictionary literal above on lines 11-16. Consider just returning the instance of the class here without first creating `new_goal` since you don't do anything with the variable before returning it:
```suggestion
        return cls(title=goal_data["title"])
```